File: pset6/credit/credit.py
from cs50 import get_int
import sys

# get card nober from user
no = get_int("number: ")
temp = no
n = 0
sum = 0
f = 0
temp2 = 0
i = 0

# Apply luhn's algorithm
while temp >= 1:
    n += 1
    # for even places
    if (n % 2) == 0:
        i = ((temp % 10) * 2)
        if i > 9:
            temp2 = 0
            for j in range(2):
                temp2 = temp2 + (i % 10)
                i /= 10
        else:
            temp2 = i
        sum += temp2
    # for odd places
    else:
        sum = sum + (temp % 10)
    temp /= 10
    temp = int(temp)
    sum = int(sum)
temp = no

# terminating program if end digit of sum is not 0
if (sum % 10) != 0:
    print("INVALID")
    sys.exit(0)

# Printing company of card. Also, the flag comes in action here
# Checking conditions for American Express
if n == 15:
    for i in range(14):
        if int(no / 10) == 34 or int(no / 10) == 37:
            print("AMEX")
            f = 1
        no /= 10
# conditions for VISA
elif n == 13 or n == 16:
    for i in range(16):
        if int(temp / 10) == 4:
            print("VISA")
            f = 1
        temp /= 10
# condition for MASTERCARD
if n == 16 and f == 0:
    for i in range(14):
        if int(no / 10) == 51 or int(no / 10) == 52 or int(no / 10) == 53 or int(no / 10) == 54 or int(no / 10) == 55:
            print("MASTERCARD")
            f = 1
        no /= 10
# Rest all cases involved
if f == 0:
    print("INVALID")

File: pset1/credit/credit.c
#include<stdio.h>
#include<stdlib.h>
#include<cs50.h>

int main(void)
{
    long int num, temp; 
    int sum = 0, n = 0, i, temp2 = 0, f = 0;//f is for flag
    
    //Getting number from user
    num = get_long("Number: ");
    temp = num;
    
// Applying Luhn's Algorithm
    
    while (temp > 0)
    {
        n++;
        //For even places
        if ((n % 2) == 0)
        {
            i = (temp % 10) * 2;
            if (i > 9)
            {
                temp2 = 0;
                for (int j = 0; j < 2; j++)
                {
                    temp2 += (i % 10);
                    i /= 10;
                }
            }
            else
            {
                temp2 = i;
            }
            sum += temp2;
        }
        //For odd Places
        else
        {
            sum += (temp % 10);
        }
        temp /= 10;
    }
    
    temp = num;
    
    //Terminating program if end digit of sum is not 0
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }
    
    //Printing company of Card. Also, the flag comes in action here.
    //Checking conditions for American Express
    if (n == 15)
    {
        for (i = 0; i <= 13; i++)
        {
            if ((num / 10) == 34 || (num / 10) == 37)
            {
                printf("AMEX\n");
                f = 1;
            }
            num /= 10;
        }
    }
    //Conditions of VISA
    else if (n == 13 || n == 16)
    {
        for (i = 0; i <= 15; i++)
        {
            if (temp / 10 == 4)
            {
                printf("VISA\n");
                f = 1;
            }
            temp /= 10;
        }
    }
    //Conditions of MASTERCARD
    if (n == 16 && f == 0)
    {
        for (i = 0; i < 14; i++)
        {
            if ((num / 10 == 51) || (num / 10 == 52) || (num / 10 == 53) || (num / 10 == 54) || (num / 10 == 55))
            {
                printf("MASTERCARD\n");
                f = 1;
            }
            num /= 10;
        }
    }
    //Rest all cases invalid.
    if (f == 0)
    {
        printf("INVALID\n");
    }
}